downstairs/src/lib.rs

```diff
@@ -1180,92 +1259,122 @@ impl Downstairs {
         uuid: Uuid,
         tx: Arc<Sender<u64>>,
     ) -> Result<()> {
-        let mut work = self.work.lock().await;
+        if self.read_only {
+            // If in read only mode, multiple read-only activations are allowed.
+            self.active_upstairs.insert(
```

Comment:
Does this handle collisions?  Meaning two of the same UUIDs connect?

Reply:
Currently this block of code does not handle collisions, and this looks a lot like buggy code. I think the fix should be storing generation number and using that to kick out the old Upstairs if a new one connects with the same UUID, meaning the comments (and RFD) have to be updated to include cases where read-only Upstairs **can** be kicked out / taken over.